api/v1/storageprofile_types.go

```diff
@@ -0,0 +1,103 @@
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.sharedFilesystemConfiguration) || has(self.sharedFilesystemConfiguration)", message="sharedFilesystemConfiguration is required once set"
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.blockPoolConfiguration) || has(self.blockPoolConfiguration)", message="blockPoolConfiguration is required once set"
+type StorageProfileSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="DeviceClass is immutable"
+	// +kubebuilder:validation:MaxLength=512
+	// DeviceClass is the deviceclass name.
+	DeviceClass string `json:"deviceClass"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="SharedFilesystemConfiguration is immutable"
+	// configurations to use for cephfilesystem.
+	SharedFilesystemConfiguration SharedFilesystemConfigurationSpec `json:"sharedFilesystemConfiguration,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="BlockPoolConfiguration is immutable"
+	// configurations to use for  profile specific blockpool.
+	BlockPoolConfiguration BlockPoolConfigurationSpec `json:"blockPoolConfiguration,omitempty"`
+}
+
+// StorageProfileStatus defines the observed state of StorageProfile
+type StorageProfileStatus struct {
+	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Phase describes the Phase of StorageProfile
+	// This is used by OLM UI to provide status information
+	// to the user
+	Phase StorageProfilePhase `json:"phase,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// StorageProfile is the Schema for the storageprofiles API
+type StorageProfile struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   StorageProfileSpec   `json:"spec,omitempty"`
+	Status StorageProfileStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// StorageProfileList contains a list of StorageProfile
+type StorageProfileList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []StorageProfile `json:"items"`
+}
+
+// StorageProfilePhase stores a StorageProfile reconciliation phase
+type StorageProfilePhase string
+
+const (
+	StorageProfilePhaseFailed StorageProfilePhase = "Failed"
+	StorageProfilePhaseInit   StorageProfilePhase = "Initializing"
+	StorageProfilePhaseReady  StorageProfilePhase = "Ready"
+)
+
+// StorageProfileMessage stores a StorageProfile reconciliation message
+type StorageProfileMessage string
+
+const (
```

Comment:
- after a full pass, I realized this message isn't for CR status and as such I didn't feel the need to have this based on it's current usage (only logging)

Reply:
@leelavg if you take a closer look, it's being used for other stuff that are crucial.

Reply:
I'm aware of that, this comment should be read in conjunction with other comment mentioning comparison of strings as errors.